packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -899,12 +671,165 @@ export const prepareVaultManagerKit = (
         getPublicFacet() {
           return this.facets.collateral;
         },
+
+        lockOraclePrices() {
+          const { state, facets } = this;
+          const { self } = facets;
+
+          // XXX: 'twould be better if collateralQuote were updated lazily when it
+          // changes by more than X%, and we use a cached value.  see #6946
+          return E.when(self.getCollateralQuote(), quote => {
+            trace(`lockPrice`, getAmountIn(quote), getAmountOut(quote));
+            // @ts-expect-error declared PriceQuote | undefined. TS thinks undefined.
+            state.lockedQuote = quote;
+            return quote;
+          });
+        },
+        async liquidateVaults(auctionPF) {
+          const { state, facets } = this;
+          const { lockedQuote, compoundedInterest } = state;
+
+          assert(factoryPowers && prioritizedVaults && zcf);
+          lockedQuote ||
+            Fail`Must have locked a quote before liquidating vaults.`;
+          assert(lockedQuote); // redundant with previous line
+
+          /** @type {import('./storeUtils.js').NormalizedDebt} */
+          // @ts-expect-error cast
+          const crKey = normalizedCrKey(lockedQuote, compoundedInterest);
+
+          trace(
+            `Liquidating vaults worse than`,
+            normalizedCr(lockedQuote, compoundedInterest),
+          );
+          const { totalDebt, totalCollateral, vaultsToLiquidate, liqSeat } =
+            prioritizedVaults.removeVaultsBelow(crKey, zcf);
+          if (!totalCollateral) {
+            return;
+          }
+
+          vaultsToLiquidate.forEach(([_, v]) => {
+            v.liquidating();
+            liquidatingVaults.add(v);
+          });
+
+          trace(' Found vaults to liquidate', liquidatingVaults.getSize());
+
+          const { userSeatPromise, deposited } = await E.when(
+            E(auctionPF).getDepositInvitation(),
+            depositInvitation =>
+              offerTo(
+                zcf,
+                depositInvitation,
+                harden({ Minted: 'Currency' }),
+                harden({
+                  give: { Collateral: totalCollateral },
+                }),
+                liqSeat,
+              ),
+          );
+
+          // This can be a long wait.
+          const [proceeds] = await Promise.all([deposited, userSeatPromise]);
+
+          trace(`VM LiqV after long wait`, proceeds);
+
+          const accounting = liquidationResults(
+            totalDebt,
+            proceeds.Minted || AmountMath.makeEmpty(debtBrand),
+          );
+
+          const removeLiquidatedVault = v => {
```

Comment:
This does more than removing.
```suggestion
          const recordLiquidation = v => {
```

Reply:
done